doc/user/csv-annotations.md

```diff
@@ -0,0 +1,152 @@
+# Code Annotations for Cluster Service Versions
+
+## Overview
+
+This document describes the semantics of Cluster Service Version (CSV) [code annotations][code_annotations_design] and lists all possible annotations.
+
+## Usage
+
+All annotations have a `+operator-sdk:gen-csv:` prefix, denoting that they're parsed while executing [`operator-sdk olm-catalog gen-csv`][sdk_cli_ref].
+
+### Paths
+
+Paths are dot-separated string hierarchies with the above prefix that map to CSV [`spec`][csv_spec] field names.
+
+Example: `+operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Pod Count"`
+
+#### customresourcedefinitions
+
+- `customresourcedefinitions`: child path token
+	-	`displayName`: string
+	- `resources`: string, in the format `"kind,version,\"name\""`, where `kind`, `version`, and `name` are fields in each CSV `resources` entry
+	- `specDescriptors`, `statusDescriptors`, `actionDescriptors`: bool, or child path token
+		- `displayName`: string
+		- `path`: string
```

Comment:
My reading of the [CSV design doc](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#your-custom-resource-definitions) is that `path` should match the JSON path of the element relative to the status/spec/action descriptors fields.

If my understanding is correct, I think we should always infer `path` and not provide an annotation to allow it to be overridden, because the inferred path is the only correct path.

Consider:
```go
type Memcached struct {
    Spec   MemcachedSpec   `json:"spec"`
    Status MemcachedStatus `json:"status"`
}
type MemcachedSpec struct {
    Pods MemcachedPods `json:"podSpecs"`
}
type MemcachedStatus struct {
    Pods MemcachedPods `json:"podStatuses"`
}
type MemcachedPods struct {
    Size int32 `json:"size"`
}
```

With this structure, we would set `MemcachedPods` `path` to :
- `podSpecs.size` in `specDescriptors`
- `podStatuses.size` in `statusDescriptors`

And if the developer updates `Size` to `PodSize` (and JSON tag `size` to `podSize`) in `MemcachedPods`, the CSV generator would automatically pick that up and change to:
- `podSpecs.podSize` in `specDescriptors`
- `podStatuses.podSize` in `statusDescriptors`

If we allowed an annotation to change these paths to any arbitrary value, the UI would not be able to locate their values.

Reply:
This seems like a reasonable assumption, looking at the CSV design doc. This also reduces the number of annotations allowed, which is good imo.